sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -74,13 +74,14 @@ public class AzurePipelinesCredential implements TokenCredential {
                 request.setHeader(HttpHeaderName.AUTHORIZATION, "Bearer " + systemAccessToken);
                 request.setHeader(HttpHeaderName.CONTENT_TYPE, "application/json");
                 try (HttpResponse response = pipeline.sendSync(request, Context.NONE)) {
+                    String responseBody = response.getBodyAsBinaryData().toString();
                     if (response.getStatusCode() != 200) {
                         throw LOGGER.logExceptionAsError(new ClientAuthenticationException("Failed to get the client assertion token "
-                            + response.getBodyAsString().block()
+                            + responseBody
                             + System.lineSeparator()
                             + "For troubleshooting information see https://aka.ms/azsdk/java/identity/azurepipelinescredential/troubleshoot.", response));
                     }
-                    OidcTokenResponse tokenResponse = OidcTokenResponse.fromJson(JsonProviders.createReader(response.getBodyAsString().block()));
+                    OidcTokenResponse tokenResponse = OidcTokenResponse.fromJson(JsonProviders.createReader(responseBody));
```

Comment:
nit: need to use a try-with-resources to make sure `JsonProviders.createReader` doesn't leek any resources given it's closeable and is an SPI.

```java
try (JsonReader jsonReader = JsonProviders.createReader(responseBody)) {
    return OidcTokenResponse.fromJson(jsonReader).getOidcToken();
}
```